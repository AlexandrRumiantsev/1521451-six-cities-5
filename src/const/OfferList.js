export const MAX_OTHER_REVIEWS = 3;
export const INDEX_OTHER_REVIEWS = 3;
