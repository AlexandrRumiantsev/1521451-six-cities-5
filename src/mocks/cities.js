export default [
  {
    id: `1`,
    title: `Amsterdam`,
    link: `Amsterdam`,
    lat: 52.3809453943508,
    lon: 4.931309666406198
  },
  {
    id: `2`,
    title: `Hamburg`,
    link: `Hamburg`,
    lat: 53.5611,
    lon: 9.9933
  },
  {
    id: `3`,
    title: `Dusseldorf`,
    link: `Dusseldorf`
  },
  {
    id: `4`,
    title: `Brussels`,
    link: `Brussels`
  },
  {
    id: `5`,
    title: `Cologne`,
    link: `Cologne`
  },
  {
    id: `6`,
    title: `Paris`,
    link: `Paris`
  },
];
