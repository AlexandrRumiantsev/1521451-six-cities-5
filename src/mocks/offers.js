export default [
  {
    id: 1,
    title: `Beautiful & luxurious studio at great location`,
    gallary: [
      {
        id: 11,
        src: `/img/room.jpg`,
        alt: `Photo studio`
      },
      {
        id: 12,
        src: `/img/apartment-01.jpg`,
        alt: `Photo studio`
      },
      {
        id: 13,
        src: `/img/apartment-02.jpg`,
        alt: `Photo studio`
      },
      {
        id: 14,
        src: `/img/apartment-03.jpg`,
        alt: `Photo studio`
      },
      {
        id: 15,
        src: `/img/studio-01.jpg`,
        alt: `Photo studio`
      },
      {
        id: 16,
        src: `/img/apartment-01.jpg`,
        alt: `Photo studio`
      },
    ],
    premium: true,
    houseLevel: `Apartment`,
    rating: 4.8,
    propertyFeatures: [
      {
        field: `Type`,
        value: `Apartment`
      },
      {
        field: `Count`,
        value: `3 Bedrooms`
      },
      {
        field: `MaxPeople`,
        value: `Max 1 adults`
      }
    ],
    prices: [`€380`, `night`],
    insides: {
      1: `Wi-Fi`,
      2: `Heating`,
      3: `Kitchen`,
      4: `Fridge`,
      5: `Washing machine`,
      6: `Coffee machine`,
      7: `Dishwasher`,
      8: `Towels`,
      9: `Baby seat`,
      10: `Cabel TV`
    },
    meeter: {
      ava: `/img/avatar-angelina.jpg`,
      name: `Angelina`,
      description: [
        `A quiet cozy and picturesque that hides behind a a
      river by the unique lightness of Amsterdam.
      The building is green and from 18th century.`,
        `An independent House, strategically located between
      Rembrand Square and National Opera, but where the
      bustle of the city comes to rest in this alley flowery and colorful.`
      ]
    },
    map: `<iframe class="property__map map" src="https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d71844.96645242712!2d37.54726058773501!3d55.75580608664706!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x46b54afc73d4b0c9%3A0x3d44d6cc5757cf4c!2sMoscow!5e0!3m2!1sen!2sru!4v1602578493907!5m2!1sen!2sru" width="600" height="450" frameborder="0" style="border:0;" allowfullscreen="" aria-hidden="false" tabindex="0"></iframe>`
  },
  {
    id: 2,
    title: `Wood and stone place`,
    gallary: [
      {
        id: 21,
        src: `/img/apartment-01.jpg`,
        alt: `Photo studio`
      },
      {
        id: 22,
        src: `/img/room.jpg`,
        alt: `Photo studio`
      },
      {
        id: 23,
        src: `/img/apartment-02.jpg`,
        alt: `Photo studio`
      },
      {
        id: 24,
        src: `/img/apartment-03.jpg`,
        alt: `Photo studio`
      },
      {
        id: 25,
        src: `/img/studio-01.jpg`,
        alt: `Photo studio`
      },
      {
        id: 26,
        src: `/img/apartment-01.jpg`,
        alt: `Photo studio`
      },
    ],
    premium: false,
    houseLevel: `Private room`,
    rating: 3.8,
    propertyFeatures: [
      {
        field: `Type`,
        value: `Room`
      },
      {
        field: `Count`,
        value: `1 Bedrooms`
      },
      {
        field: `MaxPeople`,
        value: `Max 1 adults`
      }
    ],
    prices: [`€80`, `night`],
    insides: {
      1: `Wi-Fi`,
      2: `Heating`,
      3: `Kitchen`,
      4: `Fridge`,
      5: `Washing machine`,
      6: `Coffee machine`,
      7: `Dishwasher`,
      8: `Towels`,
      9: `Baby seat`,
      10: `Cabel TV`
    },
    meeter: {
      ava: `/img/avatar-angelina.jpg`,
      name: `Angelina`,
      description: [
        `A quiet cozy and picturesque that hides behind a a
  river by the unique lightness of Amsterdam.
  The building is green and from 18th century.`,
        `An independent House, strategically located between
  Rembrand Square and National Opera, but where the
  bustle of the city comes to rest in this alley flowery and colorful.`
      ]
    },
    map: `<iframe class="property__map map" src="https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d71844.96645242712!2d37.54726058773501!3d55.75580608664706!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x46b54afc73d4b0c9%3A0x3d44d6cc5757cf4c!2sMoscow!5e0!3m2!1sen!2sru!4v1602578493907!5m2!1sen!2sru" width="600" height="450" frameborder="0" style="border:0;" allowfullscreen="" aria-hidden="false" tabindex="0"></iframe>`
  },
  {
    id: 3,
    title: `Nice, cozy, warm big bed apartment`,
    gallary: [
      {
        id: 31,
        src: `/img/apartment-02.jpg`,
        alt: `Photo studio`
      },
      {
        id: 32,
        src: `/img/apartment-01.jpg`,
        alt: `Photo studio`
      },
      {
        id: 33,
        src: `/img/room.jpg`,
        alt: `Photo studio`
      },
      {
        id: 34,
        src: `/img/apartment-03.jpg`,
        alt: `Photo studio`
      },
      {
        id: 35,
        src: `/img/studio-01.jpg`,
        alt: `Photo studio`
      },
      {
        id: 36,
        src: `/img/apartment-01.jpg`,
        alt: `Photo studio`
      },
    ],
    premium: false,
    houseLevel: `Private room`,
    rating: 4.9,
    propertyFeatures: [
      {
        field: `Type`,
        value: `Room`
      },
      {
        field: `Count`,
        value: `5 Bedrooms`
      },
      {
        field: `MaxPeople`,
        value: `Max 4 adults`
      }
    ],
    prices: [`€180`, `night`],
    insides: {
      1: `Wi-Fi`,
      2: `Heating`,
      3: `Kitchen`,
      4: `Fridge`,
      5: `Washing machine`,
      6: `Coffee machine`,
      7: `Dishwasher`,
      8: `Towels`,
      9: `Baby seat`,
      10: `Cabel TV`
    },
    meeter: {
      ava: `/img/avatar-angelina.jpg`,
      name: `Angelina`,
      description: [
        `A quiet cozy and picturesque that hides behind a a
  river by the unique lightness of Amsterdam.
  The building is green and from 18th century.`,
        `An independent House, strategically located between
  Rembrand Square and National Opera, but where the
  bustle of the city comes to rest in this alley flowery and colorful.`
      ]
    },
    map: `<iframe class="property__map map" src="https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d71844.96645242712!2d37.54726058773501!3d55.75580608664706!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x46b54afc73d4b0c9%3A0x3d44d6cc5757cf4c!2sMoscow!5e0!3m2!1sen!2sru!4v1602578493907!5m2!1sen!2sru" width="600" height="450" frameborder="0" style="border:0;" allowfullscreen="" aria-hidden="false" tabindex="0"></iframe>`
  },
  {
    id: 4,
    title: `Canal View Prinsengracht`,
    gallary: [
      {
        id: 41,
        src: `/img/apartment-02.jpg`,
        alt: `Photo studio`
      },
      {
        id: 42,
        src: `/img/apartment-01.jpg`,
        alt: `Photo studio`
      },
      {
        id: 43,
        src: `/img/room.jpg`,
        alt: `Photo studio`
      },
      {
        id: 44,
        src: `/img/apartment-03.jpg`,
        alt: `Photo studio`
      },
      {
        id: 45,
        src: `/img/studio-01.jpg`,
        alt: `Photo studio`
      },
      {
        id: 46,
        src: `/img/apartment-01.jpg`,
        alt: `Photo studio`
      },
    ],
    premium: false,
    houseLevel: `Private room`,
    rating: 4.9,
    propertyFeatures: [
      {
        field: `Type`,
        value: `Room`
      },
      {
        field: `Count`,
        value: `5 Bedrooms`
      },
      {
        field: `MaxPeople`,
        value: `Max 4 adults`
      }
    ],
    prices: [`€180`, `night`],
    insides: {
      1: `Wi-Fi`,
      2: `Heating`,
      3: `Kitchen`,
      4: `Fridge`,
      5: `Washing machine`,
      6: `Coffee machine`,
      7: `Dishwasher`,
      8: `Towels`,
      9: `Baby seat`,
      10: `Cabel TV`
    },
    meeter: {
      ava: `/img/avatar-angelina.jpg`,
      name: `Angelina`,
      description: [
        `A quiet cozy and picturesque that hides behind a a
  river by the unique lightness of Amsterdam.
  The building is green and from 18th century.`,
        `An independent House, strategically located between
  Rembrand Square and National Opera, but where the
  bustle of the city comes to rest in this alley flowery and colorful.`
      ]
    },
    map: `<iframe class="property__map map" src="https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d71844.96645242712!2d37.54726058773501!3d55.75580608664706!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x46b54afc73d4b0c9%3A0x3d44d6cc5757cf4c!2sMoscow!5e0!3m2!1sen!2sru!4v1602578493907!5m2!1sen!2sru" width="600" height="450" frameborder="0" style="border:0;" allowfullscreen="" aria-hidden="false" tabindex="0"></iframe>`
  },
  {
    id: 5,
    title: `Nice, cozy, warm big bed apartment`,
    gallary: [
      {
        id: 51,
        src: `/img/apartment-02.jpg`,
        alt: `Photo studio`
      },
      {
        id: 52,
        src: `/img/apartment-01.jpg`,
        alt: `Photo studio`
      },
      {
        id: 53,
        src: `/img/room.jpg`,
        alt: `Photo studio`
      },
      {
        id: 54,
        src: `/img/apartment-03.jpg`,
        alt: `Photo studio`
      },
      {
        id: 55,
        src: `/img/studio-01.jpg`,
        alt: `Photo studio`
      },
      {
        id: 56,
        src: `/img/apartment-01.jpg`,
        alt: `Photo studio`
      },
    ],
    premium: false,
    houseLevel: `Private room`,
    rating: 4.9,
    propertyFeatures: [
      {
        field: `Type`,
        value: `Room`
      },
      {
        field: `Count`,
        value: `5 Bedrooms`
      },
      {
        field: `MaxPeople`,
        value: `Max 4 adults`
      }
    ],
    prices: [`€180`, `night`],
    insides: {
      1: `Wi-Fi`,
      2: `Heating`,
      3: `Kitchen`,
      4: `Fridge`,
      5: `Washing machine`,
      6: `Coffee machine`,
      7: `Dishwasher`,
      8: `Towels`,
      9: `Baby seat`,
      10: `Cabel TV`
    },
    meeter: {
      ava: `/img/avatar-angelina.jpg`,
      name: `Angelina`,
      description: [
        `A quiet cozy and picturesque that hides behind a a
  river by the unique lightness of Amsterdam.
  The building is green and from 18th century.`,
        `An independent House, strategically located between
  Rembrand Square and National Opera, but where the
  bustle of the city comes to rest in this alley flowery and colorful.`
      ]
    },
    map: `<iframe class="property__map map" src="https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d71844.96645242712!2d37.54726058773501!3d55.75580608664706!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x46b54afc73d4b0c9%3A0x3d44d6cc5757cf4c!2sMoscow!5e0!3m2!1sen!2sru!4v1602578493907!5m2!1sen!2sru" width="600" height="450" frameborder="0" style="border:0;" allowfullscreen="" aria-hidden="false" tabindex="0"></iframe>`
  }
];
