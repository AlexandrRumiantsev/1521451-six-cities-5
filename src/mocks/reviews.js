export default [
  {
    id: `001`,
    offers: 1,
    user: {
      id: `1`,
      name: `Max`,
      ava: `/img/avatar-max.jpg`
    },
    text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.
      April 2019`,
    rating: 2,
  },
  {
    id: `002`,
    offers: 1,
    user: {
      id: `2`,
      name: `Иван`,
      ava: `/img/avatar-max.jpg`
    },
    text: `Отличный вариант для семейных пар!`,
    rating: 4,
  },
];
