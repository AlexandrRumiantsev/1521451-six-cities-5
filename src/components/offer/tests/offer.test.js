import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {
  BrowserRouter
} from "react-router-dom";

import OfferItem from "../components/offer-item";
import OfferList from "../components/offer-list";

import {Offer, Offers} from "./mocks";



describe(`Render Offer`, () => {

  it(`Render OfferItem`, () => {
      const mockStore = configureStore([]);
      let store = null;

      beforeEach(() => {

        store = mockStore({});

        store.dispatch = jest.fn();

        const OfferItemTest = renderer.create(
          <BrowserRouter>
            <Provider store={store}>
              <OfferItem location={null} offer={Offer}/>
            </Provider>
          </BrowserRouter>

          ).toJSON();

        expect(OfferItemTest).toMatchSnapshot();

      });

  })

  it(`Render OfferList`, () => {

    const mockStore = configureStore([]);
      let store = null;

      beforeEach(() => {

        store = mockStore({});

        store.dispatch = jest.fn();

        const OfferListTest = renderer.create(
          <BrowserRouter>
            <Provider store={store}>
              <OfferList offers={Offers}/>
            </Provider>
          </BrowserRouter>
          ).toJSON();

        expect(OfferListTest).toMatchSnapshot();

      });

})

})
