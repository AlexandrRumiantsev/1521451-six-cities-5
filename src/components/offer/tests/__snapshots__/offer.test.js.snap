// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Offer Render OfferList 1`] = `
<article
  className="near-places__card place-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  
  <div
    className="near-places__image-wrapper place-card__image-wrapper"
  >
    <a
      href="/offer/34"
    >
      <img
        alt="Place image"
        className="place-card__image"
        height="200"
        src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg"
        width="260"
      />
    </a>
  </div>
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          485
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <a
        href="/login"
        onClick={[Function]}
      >
        <button
          className="place-card__bookmark-button place-card__bookmark-button--active button"
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            In bookmarks
          </span>
        </button>
      </a>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          className="rating__stars__value"
          style={
            Object {
              "width": "50%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Nice, cozy, warm big bed apartment
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      house
    </p>
  </div>
</article>
`;
